自己紹介画面のパスワード変更ボタン押下時に再設定メール送信画面へ遷移するよう修正

toshinori-m opened the pull request:
### 概要
- 自己紹介を表示する画面（TBasicSettingEditPage）について、パスワード変更ボタン追加と登録ボタン改善を行いました。
### 変更内容（パスワード変更ボタン追加、登録ボタン改善）
- redirectToSendEmail 関数の中身を、console.log から navigate("/send-email") に修正しました。
- パスワード変更ボタンをクリックすると、/send-email に遷移してパスワード再設定メール送信画面を表示します。
- 今回の修正は、後続タスク予定だった遷移処理の実装を先行して対応したものです。
- isFormValid 関数を追加し、必要な項目が未入力の場合に <Button> を disabled にするよう制御。
- btn-disabled クラスを Tailwind カスタムスタイルに追加し、非活性状態でも統一された見た目・挙動となるよう調整。
- ユーザーが未入力の状態で送信しようとしても混乱しないよう、見た目と操作性を改善。
### 補足（登録ボタン改善）
- 他のページの登録ボタン改善については別issueにて行います。

close https://github.com/toshinori-m/stay_connect/issues/221

toshinori-m committed:
fix: パスワード変更ボタン押下時に再設定メール送信画面へ遷移するよう修正

hirosi1900day commented on frontend-react/src/pages/BasicSettingEditPage.tsx:
```diff
@@ -364,7 +375,7 @@ export default function BasicSettingEditPage() {
           {ErrorList([...(formState.error ? [formState.error] : []), ...errors, ...actionState.errors])}
           <div className="text-center my-5">
-            <Button type="submit" variant="primary" size="sm" className="mr-4">更新</Button>
+            <Button type="submit" variant="primary" size="sm" className="mr-4" disabled={!isFormValid()}>更新</Button>
```
現在の実装では、isFormValid によってフォームがすべて正しく入力されるまで送信ボタンが無効（disabled）になりますが、これには以下のようなデメリットがあります。

❗ 課題点
- なぜボタンが押せないのか分かりにくい
エラーが表示されていない状態でボタンが非活性だと、ユーザーは「バグっているのでは？」と誤解してしまう可能性があります。

- 修正箇所が複数あると確認しづらい
全てのフィールドを正しく埋めるまで送信できないため、どこにエラーがあるのかを逐一確認する必要があり、項目数が多い場合はユーザー体験が悪化します。

- アクセシビリティの問題
disabled ボタンはキーボード操作やスクリーンリーダーで無視されやすく、「送信できない UI」と認識されてしまう可能性があります。

そのため
元の実装のまま常に送信ボタンは有効にする

フォーム送信時にバリデーションを行い、エラーがある場合には明示的にエラーメッセージを表示する方が
ユーザーがどこを修正すればよいか明確になり、ストレスの少ない操作が可能になるかなと思います


一方で、フォーム送信中に多重送信を防ぐ目的で disabled を使用するのはいいと思うのでそのような使い方をするのが良いのではないかなと思いますがいかがでしょうか？

送信処理が始まったタイミングでボタンを一時的に非活性化し、完了または失敗時に再度有効にする

ユーザーの意図しない連続クリックを防止し、冪等性のない処理でも安全に扱えます

toshinori-m replied:
ありがとうございます。

この修正を行なっている時、この修正でよくなっているものか？不便になりそうなものか？迷っていました。
もし今回の修正で良ければ全てのページに反映します

よろしくお願いします！

hirosi1900day commented on frontend-react/src/pages/BasicSettingEditPage.tsx:
```diff
@@ -64,6 +64,17 @@ export default function BasicSettingEditPage() {
+    const isSelfIntroValid = self_introduction.trim().length >= SELF_INTRODUCTION_MIN_LENGTH && self_introduction.trim().length <= MAX_LENGTH
+  
+    return isNameValid && isEmailValid && isBirthdayValid && isSexValid && isSelfIntroValid
+  }
```
form送信時にも同じようなロジックがある。
基本は定数で管理しているので問題ないが条件を変更したくなった時に複数箇所を変えるのは大変かも
(観点として意識して欲しかったのであえてコメント)

hirosi1900day commented on frontend-react/src/pages/BasicSettingEditPage.tsx:
今回のPRとは関係ないですが気になった部分🙏
https://github.com/toshinori-m/stay_connect/blob/f7352f5005d2328f5b14c9d6efcb6921efc9f250/frontend-react/src/pages/BasicSettingEditPage.tsx#L91-L96

この辺りzodを使う形にすることが多い気がしますね！

toshinori-m replied:
ありがとうございます！

もし、今回の修正方法でよければバリデーションを使用している他のページに反映します。

toshinori-m committed:
fix: バリデーションタイミングと送信ボタンの挙動を改善

- フォーム送信時にのみバリデーションを実施し、ボタンは常に有効に変更
- 送信中は多重送信を防止するため、一時的にボタンを無効化
- isSubmitting の状態管理を finally で確実にリセット

toshinori-m committed:
refactor: バリデーションをZodに統一し、フォーム送信処理を簡素化

hirosi1900day commented on frontend-react/src/pages/BasicSettingEditPage.tsx:
```diff
@@ -66,55 +68,67 @@ export default function BasicSettingEditPage() {
-      if (newErrors.length > 0) {
-        return { errors: newErrors, formData: currentFormState }
-      }
+      setIsSubmitting(true)
```
https://ja.react.dev/reference/react/useActionState

useActionStateのisPendingを使えばこういった処理は不要そう

toshinori-m committed:
refactor: useActionStateのisPendingで更新ボタンの状態を管理
